nwb_conversion_tools/datainterfaces/spikeglxdatainterface.py

```diff
@@ -1,15 +1,63 @@
 """Authors: Cody Baker and Ben Dichter."""
+from pathlib import Path
 from spikeextractors import SpikeGLXRecordingExtractor
 
 from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
 
 
 class SpikeGLXRecordingInterface(BaseRecordingExtractorInterface):
-    """Primary data interface class for converting a NeuroscopeRecordingExtractor."""
+    """Primary data interface class for converting the SpikeGLX format."""
 
     RX = SpikeGLXRecordingExtractor
 
-    def get_metadata():
-        """Retrieve Ecephys metadata specific to the SpikeGLX format."""
-        # TODO
-        raise NotImplementedError
+    def get_metadata(self):
+        """Auto-populate as much metadata as possible from the SpikeGLX format."""
+        file_path = Path(self.input_args['file_path'])
+        session_id = file_path.parent.stem
+
+        n_shanks = int(self.recording_extractor._meta['snsShankMap'][1])
+        if n_shanks > 1:
+            raise NotImplementedError("SpikeGLX metadata for more than a single shank is not yet supported.")
+
+        channels = self.recording_extractor.get_channel_ids()
+        shank_electrode_number = channels
+        shank_group_name = ["Shank1" for x in channels]
+
+        re_metadata = dict(
+            Ecephys=dict(
+                Device=[
+                    dict(
+                        description=session_id + '.ap.meta'
+                    )
+                ],
+                ElectrodeGroup=[
+                    dict(
+                        name='Shank1',
+                        description="Shank1 electrodes."
+                    )
+                    for n in range(n_shanks)
+                ],
+                Electrodes=[
+                    dict(
+                        name='shank_electrode_number',
+                        description="0-indexed channel within a shank.",
+                        data=shank_electrode_number
+                    ),
+                    dict(
+                        name='group',
+                        description="A reference to the ElectrodeGroup this electrode is a part of.",
+                        data=shank_group_name
+                    ),
+                    dict(
+                        name='group_name',
+                        description="The name of the ElectrodeGroup this electrode is a part of.",
+                        data=shank_group_name
+                    )
```

Comment:
So `group` is a reference to ElectrodeGroup object, and `group_name` is the name of that object. I don't see a reason to pass both of these. You should be able to get the `group` value from the `group_name` value.

Reply:
Good catch, this may be a relic from prior to the add_electrode refactor, when passing both was necessary to create the link in the dynamic table. In theory just group_name should work currently, but I'll test it

Reply:
Right you are, this is from back when the process of adding electrode metadata required one key for each column of non-default information, so group and group_name had to exist separately and correspond one-to-one.

Since the refactor, though, only group_name is required and auto-populates the group link based only on that.

Removed this from both spikeglx and neuroscope.